flink-table/flink-table-planner-blink/src/main/scala/org/apache/flink/table/planner/codegen/SinkCodeGenerator.scala

```diff
@@ -102,7 +102,8 @@ object SinkCodeGenerator {
     }
 
     val consumedDataType = sink.getConsumedDataType
-    val outTerm = genToExternal(ctx, physicalOutputType, afterIndexModify)
+    // still uses the old conversion stack due to FLINK-18701
```

Comment:
We will merge this as is without first adressing FLINK-18701?

Reply:
Yes, I think we should merge this even without first addressing FLINK-18701. The issue seems tricky and has no assignee yet. But we should fix FLINK-18701 for 1.12.